packages/bson-transpilers/codegeneration/CodeGenerationVisitor.js

```diff
@@ -545,7 +564,7 @@ module.exports = (ANTLRVisitor) => class CodeGenerationVisitor extends ANTLRVisi
     if (ctx.type.argsTemplate) { // NOTE: not currently being used anywhere.
       args = visitedElements.map((arg, index) => {
         const last = !visitedElements[index + 1];
-        return ctx.type.argsTemplate(arg, ctx.indentDepth, last);
+        return ctx.type.argsTemplate.bind(this.getState())(arg, ctx.indentDepth, last);
```

Comment:
Does `bind` just merge the variables in `state` with the ones already in `this`?

Reply:
It actually replaces the existing `this` of the prototypes with `state`.  The solutions _seems_ safe based on my conversations with @mcasimir , but there may definitely be underlying problems I'm not aware of.  An alternative would be to merge the two objects before the bind.